.buildkite/scripts/backport_branch.sh

```diff
@@ -0,0 +1,195 @@
+#!/bin/bash
+
+source .buildkite/scripts/common.sh
+
+set -euo pipefail
+
+if buildkite-agent meta-data exists DRY_RUN; then
+  DRY_RUN="$(buildkite-agent meta-data get DRY_RUN)"
+else
+  DRY_RUN=${DRY_RUN:-"true"}
+fi
+
+if buildkite-agent meta-data exists BASE_COMMIT; then
+  BASE_COMMIT="$(buildkite-agent meta-data get BASE_COMMIT)"
+else
+  BASE_COMMIT=${BASE_COMMIT:-""}
+fi
+
+if buildkite-agent meta-data exists PACKAGE_NAME; then
+  PACKAGE_NAME="$(buildkite-agent meta-data get PACKAGE_NAME)"
+else
+  PACKAGE_NAME=${PACKAGE_NAME:-""}
+fi
+
+if buildkite-agent meta-data exists PACKAGE_VERSION; then
+  PACKAGE_VERSION="$(buildkite-agent meta-data get PACKAGE_VERSION)"
+else
+  PACKAGE_VERSION="${PACKAGE_VERSION:-""}"
+fi
+
+if buildkite-agent meta-data exists REMOVE_OTHER_PACKAGES; then
+  REMOVE_OTHER_PACKAGES="$(buildkite-agent meta-data get REMOVE_OTHER_PACKAGES)"
+else
+  REMOVE_OTHER_PACKAGES="${REMOVE_OTHER_PACKAGES:-"false"}"
+fi
+
+if [[ -z "$PACKAGE_NAME" ]] || [[ -z "$PACKAGE_VERSION" ]]; then
+  buildkite-agent annotate "The variables **PACKAGE_NAME** or **PACKAGE_VERSION** aren't defined, please try again" --style "warning"
+  exit 1
+fi
+
+FULL_ZIP_PACKAGE_NAME="${PACKAGE_NAME}-${PACKAGE_VERSION}.zip"
+TRIMED_PACKAGE_VERSION=""
+TRIMED_PACKAGE_VERSION="$(echo "$PACKAGE_VERSION" | cut -d '.' -f -2)"
+SOURCE_BRANCH="main"
+BACKPORT_BRANCH_NAME="backport-${PACKAGE_NAME}-${TRIMED_PACKAGE_VERSION}"
+PACKAGES_FOLDER_PATH="packages"
+
+isPackagePublished() {
+  local packageZip=$1
+  local responseCode
+  responseCode=$(curl -s -o /dev/null -w "%{http_code}" "https://package-storage.elastic.co/artifacts/packages/${packageZip}")
+  if [[ $responseCode == "200" ]]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+commitExists() {
+  local commit_sha=$1
+  local branch=$2
+  git checkout $branch
+  local searchResult=""
+  searchResult="$(git branch --contains $commit_sha | grep $branch | awk '{print $2}')"
```

Comment:
Nit. We can control the format so we don't need to parse the output. And we can grep with the regexp delimiters to ensure that the branch name matches exactly.
```suggestion
  searchResult="$(git branch --contains $commit_sha --format=%(refname:short) | grep -E ^${branch}$)"
```

Reply:
changed